src/render.js

```diff
@@ -107,3 +110,64 @@ function debug(
   debugImpl.shallow = (message) => debugShallow(instance, message);
   return debugImpl;
 }
+
+function appendFindAllTrap(renderer: ReactTestRenderer) {
```

Comment:
I would suggest removing `Proxy` code, and using `filter(isReactTestElementVisibleToAccessibility)` in the queries. This way `respectAccessibility` would be moved from `render` option to query option, which would better reflect the fact that we are apply query modifier and not render modifier.

We should have some base `QueryOptions` type that would atm be `{ respectAccessibility: boolean }` and would be passed as a second argument to all `get/getAll/query`/etc/ queries. Note, for `findBy` & `findAllBy` we would add the existing `waitForOptions` there.